### PR TITLE
removed ASPECT_PETSC

### PR DESCRIPTION
**Pull request by ahollyday**
removes ASPECT_USE_PETSC clause.

### CODE REVIEW

**Comment by bangerth** [comment]
You will have to force-push the changed branch we made. For this, go to the aspect source directory and do
```
  git push -f origin remove_unused_options
```
This overwrites the changes you have here in the pull request by the ones you currently have in the branch on your hard drive.

**Comment by bangerth** [comment]
Related to #3338.

File: source/postprocess/matrix_statistics.cc [hunk 1]
```diff
@@ -64,21 +64,7 @@ namespace
         // If the locale doesn't work, just give up
       }
 
-#ifdef ASPECT_USE_PETSC
-    // TODO: PETSc statistics, n_nonzero_elements doesn't exist.
-#else
-    output << "Total " << matrix_name << " nnz: "
```
Comment (tjhei) [inline source/postprocess/matrix_statistics.cc:70]: Keep this whole block please.

**Comment by tjhei** [comment]
@ahollyday Your email address is not set up correctly, so your commits are not attributed to you. Can you find one of the maintainers to walk you through this?

**Review by gassmoeller — APPROVED** [review]
Ok, this looks good now, I will start the automatic tester, and if it finishes this can be merged.

**Review by bangerth — APPROVED** [review]

**Pull request merged**